pkg/frontend/transport/handler.go

```diff
@@ -188,6 +190,22 @@ func (f *Handler) reportSlowQuery(r *http.Request, queryString url.Values, query
 	level.Info(util_log.WithContext(r.Context(), f.log)).Log(logMessage...)
 }
 
+func (f *Handler) reportFailedQuery(r *http.Request, queryString url.Values, queryResponseTime time.Duration, stats *querier_stats.Stats, err error) {
+	// Log failed query info
+	logMessage := append([]interface{}{
+		"msg", "failed query stats",
```

Comment:
Instead of having a different log message, what if we just reuse `reportQueryStats()` (so that even partially collected information about the query complexity may be logged) and we add a status field to the log to track whether the query succeeded or failed? /cc @aknuds1 @56quarters @pstibrany

Reply:
I don't feel too strongly about this either way, was thinking it might be better to keep the failed query reporting quite lightweight, but if all the additional info is going to add a lot of value I can re-use `reportQueryStats()`

Reply:
What @pracucci says sounds reasonable. The benefit of re-using `reportQueryStats` is you get all the same stats for failed queries. Is there any argument against calling `reportQueryStats` for failed queries?